perun-registrar-lib/src/test/java/cz/metacentrum/perun/registrar/RegistrarBaseIntegrationTest.java

```diff
@@ -1259,6 +1260,29 @@ public void testRejectApplicationsAfterMemberRemoval() throws PerunException {
 		assertEquals(1, group1Apps.size());
 	}
 
+	@Test
+	public void testRejectApplications() throws PerunException {
+		User user = new User(2, "User1", "Test1", "", "", "");
+		user = perun.getUsersManagerBl().createUser(session, user);
+
+		Vo otherVo = perun.getVosManager().createVo(session, new Vo(-1,"otherRegistrarTestVO","otherRegTestVO"));
+
+		Application application1 = prepareApplicationToVo(user);
+		application1 = registrarManager.submitApplication(session, application1, new ArrayList<>());
+
+		Application application2 = prepareApplicationToVo(user);
+		application2.setVo(otherVo);
+		application2 = registrarManager.submitApplication(session, application2, new ArrayList<>());
+
+		registrarManager.rejectApplications(session, List.of(application1.getId(), application2.getId()), null);
+
+		List<Integer> rejectedAppIdsVO = registrarManager.getApplicationsForVo(session, vo, List.of("REJECTED"), false).stream().map(Application::getId).toList();
+		assertThat(rejectedAppIdsVO).containsExactly(application1.getId());
+
+		List<Integer> rejectedAppIdsOtherVo = registrarManager.getApplicationsForVo(session, otherVo, List.of("REJECTED"), false).stream().map(Application::getId).toList();
+		assertThat(rejectedAppIdsOtherVo).containsExactly(application2.getId());
+	}
```

Comment:
I also considered setting another vo for the second application (when we had a call about it before), but this is not the case in how it will be used - the manager will be under one vo and he/she will want to reject several applications under this one vo.

I tried to look more deeply into it and what do you think about this suggestion:
Because there is a problem just with the principal, that filter method in `getOpenApplicationsForUserInVo()` method evaluates the second application as related to the same user (according to some check of actor and extSourceName), what about just creating user2, another session2 and create a relation between this session2 (`session2.getPerunPrincipal().getActor()`) and application2 (related to user2).
```suggestion
	public void testRejectApplications() throws PerunException {
		User user = new User(1, "User1", "Test1", "", "", "");
		user = perun.getUsersManagerBl().createUser(session, user);

		User user2 = new User(2, "User2", "Test2", "", "", "");
		user2 = perun.getUsersManagerBl().createUser(session, user2);

		Application application1 = prepareApplicationToVo(user);
		application1 = registrarManager.submitApplication(session, application1, new ArrayList<>());

        PerunSession session2 = perun.getPerunSession(
			new PerunPrincipal("perunTests2", ExtSourcesManager.EXTSOURCE_NAME_INTERNAL, 
        ExtSourcesManager.EXTSOURCE_INTERNAL),
			new PerunClient());
		Application application2 = prepareApplicationToVo(user2);
		application2.setCreatedBy(session2.getPerunPrincipal().getActor());
		application2 = registrarManager.submitApplication(session, application2, new ArrayList<>());

		registrarManager.rejectApplications(session, List.of(application1.getId(), application2.getId()), null);

		List<Integer> rejectedAppIdsVO = registrarManager.getApplicationsForVo(session, vo, List.of("REJECTED"), false).stream().map(Application::getId).toList();
		assertThat(rejectedAppIdsVO).containsExactlyInAnyOrder(application1.getId(), application2.getId());
	}
```

Reply:
In this case, we don't need to create a whole new session. Just set the value in `createdBy` to something different, right?

Reply:
Even easier way :raised_hands: